PlayerKit/Classes/RegularPlayer.swift

```diff
@@ -400,3 +408,57 @@ extension RegularPlayer: FillModeCapable
         }
     }
 }
+
+extension RegularPlayer: TextTrackCapable
+{
+    public func availableTextTracks() -> [TextTrackMetadata]
+    {
+        guard let group = self.player.currentItem?.asset.mediaSelectionGroup(forMediaCharacteristic: .legible) else
+        {
+            return []
+        }
+        return group.options
+    }
+    
+    public func fetchAvailableTextTracks(completion: @escaping ([TextTrackMetadata]) -> Void)
+    {
+        self.player.currentItem?.asset.loadValuesAsynchronously(forKeys: [#keyPath(AVAsset.availableMediaCharacteristicsWithMediaSelectionOptions)]) { [weak self] in
+            guard let strongSelf = self, let group = strongSelf.player.currentItem?.asset.mediaSelectionGroup(forMediaCharacteristic: .legible) else
+            {
+                completion([])
+                return
+            }
+            completion(group.options)
+        }
+    }
+    
+    public func select(_ textTrack: TextTrackMetadata?)
+    {
+        guard let group = self.player.currentItem?.asset.mediaSelectionGroup(forMediaCharacteristic: .legible) else
+        {
+            return
+        }
+        
+        guard let track = textTrack else
+        {
+            self.player.currentItem?.select(nil, in: group)
+            return
+        }
+        
+        if let option = track as? AVMediaSelectionOption
+        {
+            self.player.currentItem?.select(option, in: group)
+        }
+        else
+        {
+            for option in group.options
```

Comment:
I don't think we have a guideline for this, but we tend to prefer operating on collections in closures. It's not required but something like this:

```Swift
group.options.forEach { [weak self] (option) in
    guard let strongSelf = self,
        option.locale == track.locale && (option.hasMediaCharacteristic(.describesMusicAndSoundForAccessibility) == track.describesMusicAndSound) else
    {
            return
    }
    strongSelf.player.currentItem?.select(option, in: group)
}
```

I guess I don't know how I feel about my guard statement, haha. It's kind of ugly 🙀